### PR TITLE
Run common checks for every single solution; Add common check for snake_case

### PR DESCRIPTION
**Pull request by angelikatyborska**
This is my attempt at adding common checks that will be done on every single solution, as suggested in https://github.com/exercism/elixir-analyzer/issues/97

For starters, I'm adding two checks that were pretty easy to implement:
- module attribute names should be snake_case
- function names should be snake_case

I can also see that it should be relatively easy to implement a check for PascalCase module names. It might be a bit more difficult to check for snake_case variable names, but still worth it.

I would also want to later add a check for not using tabs for indentation, but only as a `informative` comment not to be too annoying.

Some checks can be "stolen" from Credo :)
- module names: https://github.com/rrrene/credo/blob/master/lib/credo/check/readability/module_names.ex
- snake case variable names: https://github.com/rrrene/credo/blob/master/lib/credo/check/readability/variable_names.ex
- predicate function names (for Bob!): https://github.com/rrrene/credo/blob/master/lib/credo/check/readability/predicate_function_names.ex
- `@impl ModuleName` instead of `@impl true`: https://github.com/rrrene/credo/blob/master/lib/credo/check/readability/impl_true.ex

Website copy PR: https://github.com/exercism/website-copy/pull/2019

### CODE REVIEW

**Comment by angelikatyborska** [comment]
@jiegillet your review would also be much appreciated!

File: lib/elixir_analyzer/test_suite/default.ex [hunk 1]
```diff
@@ -0,0 +1,10 @@
+defmodule ElixirAnalyzer.TestSuite.Default do
+  @moduledoc """
+  This is the default exercise analyzer extension module.
+
+  It will be run for any exercise submission that doesn't have its own extension module.
+  It's empty, which means it will only run the common checks.
```
Comment (neenjaw) [inline lib/elixir_analyzer/test_suite/default.ex:6]:
```suggestion
  It's intentionally empty, which means it will only run the common checks.
```

**Review by neenjaw — APPROVED** [review]

File: lib/elixir_analyzer/exercise_test/common_checks/function_names.ex [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    if names == [] do
+      []
+    else
+      wrong_name =
+        names
+        |> Enum.reverse()
+        |> hd()
+        |> to_string()
+
```
Comment (neenjaw) [inline lib/elixir_analyzer/exercise_test/common_checks/function_names.ex:28]: Could be simplified a little

```suggestion
    wrong_name = List.last(names)
    
    if wrong_name do
      wrong_name = to_string(wrong_name)

```

File: lib/elixir_analyzer/exercise_test/common_checks/function_names.ex [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  end
+
+  defp snake_case?(name) do
+    # the code had to compile and pass all the tests to get to the analyzer
+    # so we can assume the name is otherwise valid
+    to_snake_case(name) == to_string(name)
+  end
+
+  defp to_snake_case(name) do
+    # Macro.underscore is good enough because a module attribute name must be a valid Elixir identifier anyway
```
Comment (neenjaw) [inline lib/elixir_analyzer/exercise_test/common_checks/function_names.ex:64]: 100% agree, despite the warning on the function not to use, I see it used everywhere in non-core code :P

File: lib/elixir_analyzer/exercise_test/common_checks/module_attribute_names.ex [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    {_, names} = Macro.prewalk(ast, [], &traverse/2)
+
+    if names == [] do
+      []
+    else
+      wrong_name =
+        names
+        |> Enum.reverse()
+        |> hd()
+        |> to_string()
```
Comment (neenjaw) [inline lib/elixir_analyzer/exercise_test/common_checks/module_attribute_names.ex:25]: Same simplification could be done here

File: lib/elixir_analyzer/exercise_test/common_checks/function_names.ex [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+         %Comment{
+           type: :actionable,
+           comment: Constants.solution_function_name_snake_case(),
+           params: %{
+             expected: correct_name,
+             actual: wrong_name
+           }
+         }}
+      ]
+    end
```
Comment (neenjaw) [inline lib/elixir_analyzer/exercise_test/common_checks/function_names.ex:42]: Could be simplified a little

```suggestion
    wrong_name = List.last(names)
    
    if wrong_name do
      wrong_name = to_string(wrong_name)
      correct_name = to_snake_case(wrong_name)

      [
        {:fail,
         %Comment{
           type: :actionable,
           comment: Constants.solution_function_name_snake_case(),
           params: %{
             expected: correct_name,
             actual: wrong_name
           }
         }}
      ]
    else
      []
    end
```

Comment (angelikatyborska) [reply lib/elixir_analyzer/exercise_test/common_checks/function_names.ex]: `List.last`, why didn't I think of that 😂

**Comment by angelikatyborska** [comment]
I'll merge to see it in action and add more checks in new PR(s)

**Pull request merged**